index.go

```diff
@@ -140,24 +151,23 @@ func (idx *faissIndex) Add(x []float32) error {
 	return nil
 }
 
-func (idx *faissIndex) ObtainClusterToVecIDsFromIVFIndex() (map[int64][]int64, error) {
-	// This type assertion is required to determine whether to invoke
-	// ObtainClustersWithDistancesFromIVFIndex, SearchClustersFromIVFIndex or not.
-	if ivfIdx := C.faiss_IndexIVF_cast(idx.cPtr()); ivfIdx == nil {
-		return nil, nil
+func (idx *faissIndex) ObtainClusterVectorCountsFromIVFIndex(vecIDs []int64) (map[int64]int64, error) {
+	if !idx.IsIVFIndex() {
+		return nil, fmt.Errorf("index is not an IVF index")
 	}
-
-	clusterVectorIDMap := make(map[int64][]int64)
-
-	nlist := C.faiss_IndexIVF_nlist(idx.idx)
-	for i := 0; i < int(nlist); i++ {
-		list_size := C.faiss_IndexIVF_get_list_size(idx.idx, C.size_t(i))
-		invlist := make([]int64, list_size)
-		C.faiss_IndexIVF_invlists_get_ids(idx.idx, C.size_t(i), (*C.idx_t)(&invlist[0]))
-		clusterVectorIDMap[int64(i)] = invlist
+	rv := make(map[int64]int64, len(vecIDs))
+	for _, vecID := range vecIDs {
+		clusterID := C.faiss_get_list_for_key(idx.idx, (C.idx_t)(vecID))
+		rv[int64(clusterID)] += 1
```

Comment:
nit: just a coding convention thing - its better to use `rv[int64(clusterID)]++`